src/Raven.Server/RavenServer.cs

```diff
@@ -1143,26 +1143,56 @@ private void ListenToNewTcpConnection(TcpListener listener)
         {
             Task.Run(async () =>
             {
+                var backoffSecondsDelay = 0;
                 TcpClient tcpClient;
-                Logger tcpAuditLog = null;
-                try
-                {
-                    tcpClient = await listener.AcceptTcpClientAsync();
-                    tcpAuditLog = LoggingSource.AuditLog.IsInfoEnabled ? LoggingSource.AuditLog.GetLogger("TcpConnections", "Audit") : null;
-                }
-                catch (ObjectDisposedException)
-                {
-                    // shutting down
-                    return;
-                }
-                catch (Exception e)
+                Logger tcpAuditLog;
+
+                while (true)
                 {
-                    if (_tcpLogger.IsInfoEnabled)
+                    try
                     {
-                        _tcpLogger.Info("Failed to accept new tcp connection", e);
+                        tcpClient = await listener.AcceptTcpClientAsync();
+                        tcpAuditLog = LoggingSource.AuditLog.IsInfoEnabled ? LoggingSource.AuditLog.GetLogger("TcpConnections", "Audit") : null;
+                        break;
                     }
+                    catch (ObjectDisposedException)
+                    {
+                        // shutting down
+                        return;
+                    }
+                    catch (Exception e)
+                    {
+                        if (backoffSecondsDelay == 0)
+                        {
+                            if (_tcpLogger.IsInfoEnabled)
+                            {
+                                _tcpLogger.Info($"Failed to accept new tcp connection, will retry now", e);
+                            }
 
-                    return;
+                            backoffSecondsDelay = 1;
+                            continue;
+                        }
+
+                        if (_tcpLogger.IsOperationsEnabled)
+                        {
+                            _tcpLogger.Operations($"Failed to accept new tcp connection again, will wait {backoffSecondsDelay} seconds before retrying", e);
+                        }
+
+                        await Task.Delay(TimeSpan.FromSeconds(backoffSecondsDelay), ServerStore.ServerShutdown);
```

Comment:
this can throw OperationCanceledException and/or ObjectDisposedException probably?

Reply:
We want to handle `OperationCanceledException ` as well, yes

Reply:
Not sure why this is necessary... we are shutting down in that case, but done.